cpp/Platform.Random/RandomExtensions.h

```diff
@@ -1,11 +1,17 @@
 ﻿namespace Platform::Random
 {
-    class RandomExtensions
+    static std::uint64_t NextUInt64(const std::mt19937_64& random)
     {
-        public: static std::uint64_t NextUInt64(System::Random random) { return random.NextUInt64(Range.std::uint64_t); }
+        return NextUInt64(random, Range::UInt64);
+    }
 
-        public: static std::uint64_t NextUInt64(System::Random random, Range<std::uint64_t> range) { return (std::uint64_t)(random.NextDouble() * range.Difference()) + range.Minimum; }
+    static std::uint64_t NextUInt64(const std::mt19937_64& random, const Range<std::uint64_t>& range)
+    {
+        return std::uniform_int_distribution<std::uint64_t>(range.Minimum, range.Maximum)(random);
+    }
 
-        public: static bool NextBoolean(System::Random random) { return random.Next(2) == 1; }
-    };
-}
\ No newline at end of file
+    static bool NextBoolean(const std::mt19937_64& random)
+    {
+        return std::uniform_int_distribution<int>(0, 1)(random);
+    }
```

Comment:
Similarly to do:
```cpp
static std::uint64_t NextUInt64(std::mt19937_64& random)
{
    static const std::uniform_int_distribution<std::uint64_t> generator(0, 1);
    return generator(random);
}
```

Reply:
Also might use `auto&`
Since the operator() from `std::uniform_int_distribution` looks like this:
```cpp
template<typename _UniformRandomBitGenerator>
result_type operator()(_UniformRandomBitGenerator& __urng);
```